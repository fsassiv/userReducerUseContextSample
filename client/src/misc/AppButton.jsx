import React from "react";
import { Button } from "@material-ui/core";

function AppButton({ btnLabel, btnHandleClick }) {
  return (
    <Button
      variant="contained"
      color="primary"
      display="block"
      disableElevation
      onClick={btnHandleClick}
      style={{ height: "40px", fontWeight: "lighter", width: "100%" }}
    >
      {btnLabel}
    </Button>
  );
}

export default AppButton;
