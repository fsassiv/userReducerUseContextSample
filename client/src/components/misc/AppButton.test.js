import AppButton from "./AppButton.jsx";

describe("AppButton", () => {
  const mockClick = jest.fn();
  const props = { btnLabel: "Click Me", btnHandleClick: mockClick };
  const appBtn = shallow(<AppButton {...props} />);

  it("rendered correctly", () => {
    expect(appBtn).toMatchSnapshot();
  });

  describe("when clicked, call the 'btnHandleClick' function", () => {
    beforeEach(() => {
      appBtn.find(".appBtn").simulate("click");
    });

    it("call the 'btnHandleClick' callback", () => {
      expect(mockClick).toHaveBeenCalled();
    });
  });
});
