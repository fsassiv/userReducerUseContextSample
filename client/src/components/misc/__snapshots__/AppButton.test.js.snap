// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton rendered correctly 1`] = `ShallowWrapper {}`;
